In [3]:
import numpy as np
import pandas as pd

In [4]:
address = 'C:\\Users\Charbel\\Desktop\\Udemy\\2021 Data Science - MachineLearning Project Deployment Mastery\\Automobile.csv'

df = pd.read_csv(address)
df.head()

symboling  normalized_losses         make fuel_type aspiration  \
0          3                168  alfa-romero       gas        std   
1          3                168  alfa-romero       gas        std   
2          1                168  alfa-romero       gas        std   
3          2                164         audi       gas        std   
4          2                164         audi       gas        std   

  number_of_doors   body_style drive_wheels engine_location  wheel_base  ...  \
0             two  convertible          rwd           front        88.6  ...   
1             two  convertible          rwd           front        88.6  ...   
2             two    hatchback          rwd           front        94.5  ...   
3            four        sedan          fwd           front        99.8  ...   
4            four        sedan          4wd           front        99.4  ...   

   engine_size  fuel_system  bore  stroke compression_ratio horsepower  \
0          130         mpfi  3.47    2.68               9.0        111   
1          130         mpfi  3.47    2.68               9.0        111   
2          152         mpfi  2.68    3.47               9.0        154   
3          109         mpfi  3.19    3.40              10.0        102   
4          136         mpfi  3.19    3.40               8.0        115   

   peak_rpm city_mpg  highway_mpg  price  
0      5000       21           27  13495  
1      5000       21           27  16500  
2      5000       19           26  16500  
3      5500       24           30  13950  
4      5500       18           22  17450  

[5 rows x 26 columns]

In [5]:
df.isnull().sum()

symboling              0
normalized_losses      0
make                   0
fuel_type              0
aspiration             0
number_of_doors        0
body_style             0
drive_wheels           0
engine_location        0
wheel_base             0
length                 0
width                  0
height                 0
curb_weight            0
engine_type            0
number_of_cylinders    0
engine_size            0
fuel_system            0
bore                   0
stroke                 0
compression_ratio      0
horsepower             0
peak_rpm               0
city_mpg               0
highway_mpg            0
price                  0
dtype: int64

#### No NaN items observed in the dataset

In [7]:
df.shape

(201, 26)

#### 201 Rows by 26 Columns

In [9]:
df_pricey=df[df['price']>30000][['make','price']]
df_pricey

make  price
14             bmw  30760
15             bmw  41315
16             bmw  36880
44          jaguar  32250
45          jaguar  35550
46          jaguar  36000
67   mercedes-benz  31600
68   mercedes-benz  34184
69   mercedes-benz  35056
70   mercedes-benz  40960
71   mercedes-benz  45400
123        porsche  32528
124        porsche  34028
125        porsche  37028

#### There are 14 cars that are priced above $30000

In [10]:
df_weak=df[df['horsepower']<100][['make','horsepower']]
df_weak

make  horsepower
17    chevrolet          48
18    chevrolet          70
19    chevrolet          70
20        dodge          68
21        dodge          68
..          ...         ...
183  volkswagen          68
185  volkswagen          90
186  volkswagen          90
188  volkswagen          68
189  volkswagen          88

[109 rows x 2 columns]

#### There are 109 cars that have less than 100 hp

In [11]:
df['price'].mean()

13207.129353233831

#### The average price of all the listed cars is $13207.13

In [15]:
df['make'].value_counts()

toyota           32
nissan           18
mazda            17
mitsubishi       13
honda            13
subaru           12
volkswagen       12
peugot           11
volvo            11
dodge             9
mercedes-benz     8
bmw               8
plymouth          7
saab              6
audi              6
porsche           4
alfa-romero       3
chevrolet         3
jaguar            3
isuzu             2
renault           2
mercury           1
Name: make, dtype: int64

#### The 3 most common cars found in the dataset are Toyota (32 cars), Nissan (18 cars) and Mazda (17 cars).